src/main/java/org/eln2/mc/common/blocks/cell/WireCellBlock.kt

```diff
@@ -1,11 +1,209 @@
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_east, true))
+                }
+                if(direction?.equals(Direction.NORTH) == true) {
+                    println("North")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_north, true))
+                }
+                if(direction?.equals(Direction.SOUTH) == true) {
+                    println("South")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_south, true))
+                }
+                if(direction?.equals(Direction.WEST) == true) {
+                    println("West")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_west, true))
+                }
+
+
+
+
+            } else {
+                // not a cell tile entity
+
+                val direction = Direction.fromNormal(neighbor.subtract(pos))
+
+                println("Direction: $direction")
+
+
+                if(direction?.equals(Direction.EAST) == true) {
+
+                    println("East")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_east, false))
+                }
+                if(direction?.equals(Direction.NORTH) == true) {
+                    println("North")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_north, false))
+                }
+                if(direction?.equals(Direction.SOUTH) == true) {
+                    println("South")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_south, false))
+                }
+                if(direction?.equals(Direction.WEST) == true) {
+                    println("West")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_west, false))
+                }
+
+            }
+
+        } else {
+            // not a cell tile entity
+
+            val direction = Direction.fromNormal(neighbor.subtract(pos))
+
+            println("Direction: $direction")
+
+
+            if(direction?.equals(Direction.EAST) == true) {
+
+                println("East")
+                level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_east, false))
+            }
+            if(direction?.equals(Direction.NORTH) == true) {
+                println("North")
+                level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_north, false))
+            }
+            if(direction?.equals(Direction.SOUTH) == true) {
+                println("South")
+                level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_south, false))
+            }
+            if(direction?.equals(Direction.WEST) == true) {
+                println("West")
+                level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_west, false))
+            }
+        }
```

Comment:
These two are the same logic, and should be replaced with a private function if possible.